src/ttrss_api.cpp

```diff
@@ -192,21 +214,13 @@ std::vector<tagged_feedurl> ttrss_api::get_subscribed_urls() {
 		}
 
 		for (json& feed : feedlist) {
-			const int feed_id            = feed["id"];
-			const std::string feed_title = feed["title"];
-			const std::string feed_url   = feed["feed_url"];
-			const int cat_id             = feed["cat_id"];
-
-			std::vector<std::string> tags;
-			tags.push_back(std::string("~") + feed_title);
-
-			if (cat_id > 0) {
-				tags.push_back(category_names[cat_id]);
-			}
-
-			auto url = strprintf::fmt("%s#%d", feed_url, feed_id);
-			feeds.push_back(tagged_feedurl(url, tags));
-			// TODO: cache feed_id -> feed_url (or feed_url -> feed_id ?)
+			feeds.push_back(feed_from_json(feed, [&](const json& f){
+					const int cat_id = f["cat_id"];
```

Comment:
This body can be moved out of `push_back` into the body of the loop; then, you can pass `feed_from_json` the result. The code becomes more straightforward, and we avoid scaring people with lambdas :)

Reply:
Good catch, the lambda is now eliminated! I guess, too much functional programming leads to spurious visions of lambdas from time to time :)